src/ngraph/runtime/cpu/pass/cpu_fusion.cpp

```diff
@@ -2519,3 +2521,69 @@ void ngraph::runtime::cpu::pass::CPUQuantFusion::construct_qconvb_add()
     auto m = std::make_shared<pattern::Matcher>(prelu, "CPUQuantFusion.QConvBiasSignedAdd");
     this->add_matcher(m, callback);
 }
+
+// Convert a QuantizedDot which takes [m,n]*[n,k] to
+// QuantizedMatmul which reorders input1 and does [m,n]*[k,n]
+// which is what mkldnn wants
+void ngraph::runtime::cpu::pass::CPUQuantFusion::construct_quantized_matmul(bool requantize,
+                                                                            bool with_relu)
+{
+    Shape shape_input0{2, 3};
+    Shape shape_input1{3, 4};
+    auto input0 = std::make_shared<pattern::op::Label>(element::u8, shape_input0);
+    auto input1 = std::make_shared<pattern::op::Label>(element::i8, shape_input1);
+    auto scale = std::make_shared<pattern::op::Label>(element::f32, Shape{});
+
+    std::shared_ptr<ngraph::op::Op> q_dot;
+    if (!requantize && !with_relu)
+    {
+        q_dot = std::make_shared<ngraph::op::QuantizedDot>(input0, input1, scale, false, false);
+    }
+    else if (!requantize && with_relu)
+    {
+        q_dot = std::make_shared<ngraph::op::QuantizedDot>(input0, input1, scale, false, true);
+    }
+    else if (requantize && !with_relu)
+    {
+        q_dot = std::make_shared<ngraph::op::QuantizedDot>(input0, input1, scale, true, false);
+    }
+    else
+    {
+        q_dot = std::make_shared<ngraph::op::QuantizedDot>(input0, input1, scale, true, true);
+    }
```

Comment:
you can use a single pattern for all the four variants of QuantizedDot since the pattern matcher only looks for op types and not the attributes.